### ISSUE TITLE
ChefDeprecations/NodeMethodsInsteadofAttributes misfire

### ISSUE THREAD

**Issue by jaymzh** [chef/cookstyle#657]
# Version:

```
Cookstyle 6.3.4
  * RuboCop 0.82.0
```

# Environment:

Ubuntu 18.04

# Scenario:
Make a test cookbook with:

libraries/t.rb
```
class Chef
  class Node
    def root_user
      'root'
    end

    def root_group
      'root'
    end
  end
end
```
and recipes/default.rb:
```
file '/tmp/test' do
  user node.root_user
  group node.root_group
end
```

and run cookstyle against it and it will not like `node.root_group` thinking you're doing attr access (which is incorrect) even though it is OK with `node.root_user`:

```
recipes/default.rb:9:14: W: ChefDeprecations/NodeMethodsInsteadofAttributes: Use node attributes to access Ohai data instead of node methods, which were deprecated in Chef Infra Client 13.
  group node.root_group
             ^^^^^^^^^^
```

Wat.

# Steps to Reproduce:

see above

# Expected Result:

it doesn't complain about node methods

# Actual Result:

it complains.

**Comment by tas50** [comment]
This cop does its best given the nature of processing AST and can't really account for people that might extend the Node class. This is just a cop that's going to need to be turned off if you're doing an advanced use case like this. For 99.9% of users they won't encounter issues with this so I'm going to close this one out.
